### PR TITLE
Skip balance when colocate group is balancing

### PR DESCRIPTION
**Pull request by kangkaisen**
Fix colocate meta error in  [546](https://github.com/apache/incubator-doris/issues/546)

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/clone/ColocateTableBalancer.java [hunk 1]
```diff
@@ -196,6 +196,11 @@ private synchronized void tryBalanceWhenBackendChange() {
 
         Set<Long> allGroupIds = colocateIndex.getAllGroupIds();
         for (Long groupId : allGroupIds) {
+            if (colocateIndex.isGroupBalancing(groupId)) {
+                LOG.warn("colocate group {} is balancing", groupId);
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/clone/ColocateTableBalancer.java]: Change the log level from 'warn' to 'info'

File: fe/src/main/java/org/apache/doris/clone/ColocateTableBalancer.java [hunk 2]
```diff
@@ -587,6 +592,8 @@ private void balanceForBackendAdded(Long groupId, Database db, List<Long> addedB
                                 backends.add(cloneReplicaBackendId);
                             }
 
+                            Preconditions.checkState(replicateNum == backends.size());
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/clone/ColocateTableBalancer.java]:
```suggestion
                            Preconditions.checkState(replicateNum == backends.size(), replicateNum + " vs. " + backends.size());
```
Better for debug

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 3]
```diff
@@ -4033,6 +4033,8 @@ private void createTablets(String clusterName, MaterializedIndex index, ReplicaS
                     Replica replica = new Replica(replicaId, backendId, replicaState, version, versionHash);
                     tablet.addReplica(replica);
                 }
+
+                Preconditions.checkState(chosenBackendIds.size() == replicationNum);
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/catalog/Catalog.java]:
```suggestion
                Preconditions.checkState(chosenBackendIds.size() == replicationNum, chosenBackendIds.size() + " vs. "+ replicationNum);
```

**Review by morningman — CHANGES_REQUESTED** [review]

File: fe/src/main/java/org/apache/doris/clone/ColocateTableBalancer.java (hunk 1 above)
Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/clone/ColocateTableBalancer.java]: OK

File: fe/src/main/java/org/apache/doris/clone/ColocateTableBalancer.java (hunk 2 above)
Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/clone/ColocateTableBalancer.java]: OK

**Review by morningman — APPROVED** [review]

**Pull request merged**